recipes/system-probe.rb

```diff
@@ -0,0 +1,49 @@
+include_recipe 'datadog::dd-agent'
+
+# Set the system-probe agent service enable or disable
+agent_enable = node['datadog']['system_probe']['enabled'] ? :enable : :disable
+# Set the correct Agent startup action
+agent_start = node['datadog']['system_probe']['enabled'] ? :start : :stop
+
+#
+# Configures system-probe agent
+#
+if node['datadog']['agent6']
+  include_recipe 'datadog::_agent6_config'
+
+  system_probe_config_file = ::File.join(node['datadog']['agent6_config_dir'], 'system-probe.yaml')
+  template system_probe_config_file do
+    source 'system_probe.yaml.erb'
+    variables(
+      enabled: node['datadog']['system_probe']['enabled'],
+      sysprobe_socket: node['datadog']['system_probe']['sysprobe_socket'],
+      debug_port: node['datadog']['system_probe']['debug_port'],
+      bpf_debug: node['datadog']['system_probe']['bpf_debug']
+    )
+    owner 'dd-agent'
+    group 'root'
+    mode '640'
+    notifies :restart, 'service[datadog-agent-sysprobe]', :delayed unless agent_start == false
```

Comment:
it looks like the values of agent_start and `:start` and `stop` both of which are truthy.

I think you want `unless agent_start == :stop`

Reply:
the `agent_start` here is a little weird. It also controls the config `enabled`. You are right that a restart would happen regardless of the value of `agent_start`, but system-probe would quit if `enabled: false`, so it'll stop right after it starts.

Reply:
I could use another flag to just indicate the `enabled` in chef land, but it's kinda duplicate.

Reply:
Oh wait I don't think what I said is what you've asked......in the definition of `agent_start` at the beginning, that ternary operator takes `:start` as true and `:stop` as false.